### PR TITLE
Deploy U-SPS containers to local Kubernetes cluster via Terraform

### LINKED ISSUE #38: Deploy U-SPS containers to local Kubernetes cluster via Terraform
The goal is to replicate the functionality of the deploy.sh script using Terraform configuration files that target a local Kubernetes cluster.

### PR DESCRIPTION
**Pull request by drewm-jpl**
- Migrated all of the actions of `hysds/deploy.sh` to Terraform.
- Additionally, [the Unity OGC ADES/WPST Flask API](https://github.com/unity-sds/ades_wpst) is now being deployed as part of the U-SPS cluster using Terraform.
- Integrated auto-generating documentation for the Terraform module, the docs can be found in `terraform/README.md`.
- The `terraform/README.md` also includes instructions for setting up a Terraform development workflow which includes a CI/CD pipeline for static analysis and code formatting of Terraform files.
- Expanded the `hysds/README.md` to include instructions for using Terraform to deploy the cluster.
- Added relevant Terraform-related files to the `.gitignore`.

resolves #38

### CODE REVIEW

File: terraform/.terraform.lock.hcl [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# This file is maintained automatically by "terraform init".
```
Comment (DustinKLo) [inline terraform/.terraform.lock.hcl:1]: we should remove this file and add its pattern to `.gitignore`

**Comment by LucaCinquini** [comment]
Hi Drew,
commenting just on the hysds/README file - could you please make this changes:
o Move the command "terraform destroy" to the section "Tear down the HySDS cluster"
o Move the section "Debugging a deployment" to the "Troubleshooting" section at the end. Also, include the LOG levels which now are missing.
o In the section "Deploy via Terraform", could you add a sentence explaining that this method will use Terrafom to deploy to the Kubernetes cluster referenced in the main.tf file, which is currently the one referenced by the local ~/.kube/config file. Additionally, it will deploy in the Kubernetes namespace named in the variables.tf file

**Comment by LucaCinquini** [comment]
o Another note: currently, the WPS-T image is not built by the "build_images.sh" script in the hysds/ directory. Could you add a note in the README file that explains how that image needs to be built separately from another Unity repository?

o Also, the terraform/README file switches to bold face towards the middle of the content

**Review by LucaCinquini — COMMENTED** [review]
Hi Drew, I tried running the Terraform scripts and all pods start except for the WPS-T app:

kubernetes_service.ades-wpst-api_service: Still creating... [9m0s elapsed]
kubernetes_service.ades-wpst-api_service: Still creating... [9m10s elapsed]
kubernetes_service.ades-wpst-api_service: Still creating... [9m20s elapsed]
kubernetes_service.ades-wpst-api_service: Still creating... [9m30s elapsed]
kubernetes_service.ades-wpst-api_service: Still creating... [9m40s elapsed]
kubernetes_service.ades-wpst-api_service: Still creating... [9m50s elapsed]
╷
│ Error: context deadline exceeded
│ 
│   with kubernetes_service.ades-wpst-api_service,
│   on ades_wpst.tf line 21, in resource "kubernetes_service" "ades-wpst-api_service":
│   21: resource "kubernetes_service" "ades-wpst-api_service" {
│ 
╵
Does it start for you, and are you able to access the server at http://localhost:5000 ?

File: terraform/.terraform.lock.hcl (hunk 1 above)
Comment (drewm-jpl) [reply terraform/.terraform.lock.hcl:1]: Hi Dustin, that's one of the files I was debating putting into the `.gitignore` and then read through [this SO post](https://stackoverflow.com/questions/67963719/should-terraform-lock-hcl-be-included-in-the-gitignore-file) and kept it. But I'm happy to add it to the `.gitignore` if you'd like.

Comment (DustinKLo) [reply terraform/.terraform.lock.hcl:1]: interesting, i guess we'll keep it in for now 👍

**Comment by LucaCinquini** [comment]
Drew: I changed the WPS-T service port to 5001 because at least on my Mac port 5000 is taken by a system app:

cinquini@MT-304074 unity-sps-prototype % lsof -i:5000
COMMAND    PID     USER   FD   TYPE             DEVICE SIZE/OFF NODE NAME
ControlCe 1596 cinquini   16u  IPv4 0x8cb30c2566d234db      0t0  TCP *:commplex-main (LISTEN)
ControlCe 1596 cinquini   17u  IPv6 0x8cb30c209b4022e3      0t0  TCP *:commplex-main (LISTEN)

**Comment by LucaCinquini** [comment]
Also please note this snipper for a config map:

resource "kubernetes_config_map" "supervisord-verdi" {
  metadata {
    name      = "supervisord-verdi"
    namespace = kubernetes_namespace.unity-sps.metadata.0.name
  }
  data = {
    "supervisord.conf" = "${file("${path.module}/../hysds/verdi/supervisord.conf")}"
  }
}

The hysds/verdi directory contains supervisord.template.conf but not supervisord.conf

So either we modify the TF script, or we create that file in the hysds/verdi directory

**Comment by drewm-jpl** [comment]
Manually verified that the Hello World PGE can be submitted and successfully run as an On-Demand Job through Tosca using the most recent commit.

**Submitted:**
Screen Shot 2022-04-15 at 10 08 07 AM

**Successfully completed:**
Screen Shot 2022-04-15 at 10 09 12 AM

**Pull request merged**